<?xml version="1.0" ?>
<robot name="irbot" xmlns:xacro="http://ros.org/wiki/xacro">

  <material name="grey">
    <color rgba="0.6 0.6 0.6 1.0"/>
  </material>

  <material name="dark">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>

  <!-- Odom publisher -->
  <!-- Comment this section if you use odom from other source
  <link name="odom"/>
  
  <joint name="odom_joint" type="fixed">
    <parent link="odom"/>
    <child link="base_footprint"/>
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
  </joint>
  -->

  <link name="base_footprint"/>

  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
  </joint>

  <link name="base_link">
    <visual>
      <origin xyz="0 -0.0097 0" rpy="1.5708 0 1.5708"/>
      <geometry>
        <mesh filename="package://slam_itbdelabo_rviz/urdf/stl/main_body.stl" scale="0.01 0.01 0.01"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <collision>
      <origin xyz="0.0 0 0.0" rpy="0 0 0"/>
      <geometry>
        <box size="0.30 0.30 0.15"/>
      </geometry>
    </collision>
  </link>

  <joint name="scan_joint" type="fixed">
    <parent link="base_link"/>
    <child link="laser"/>
    <origin xyz="0.0 0 0.0" rpy="0 0 0"/>
  </joint>

  <link name="laser">
    <visual>
      <origin xyz="0.0 -0.235 0.0" rpy="0 0 4.7168"/>
      <geometry>
        <mesh filename="package://slam_itbdelabo_rviz/urdf/stl/RPLidar.stl" scale="0.01 0.01 0.01"/>
      </geometry>
      <material name="dark"/>
    </visual>

    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.0315" radius="0.055"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="imu_joint" type="fixed">
    <parent link="base_link"/>
    <child link="imu"/>
    <origin xyz="0.0 0 0.0" rpy="0 0 0"/>
  </joint>
  
  <link name="imu"/>
</robot>

